README.md

```diff
@@ -1,179 +1,170 @@
-
-Instead of "version_ruby" and "version_bundler" consider "ruby_version" and "bundler_version". This applies to file and class naming as well.
-
-### Mixins and inheritance are discouraged
-
-Share behavior by sharing objects when possible. Relatedly: Strive for DRY concepts, not DRY source code.
-
-### Initialize values, call behavior
-
-In general it's prefered to initialize all values when creating an object rather than passing values in later. We also want to decouple actions from initialization. The pattern here is to have actions respond when executintg `call` on the object. In general it allows us to be flexible with when we create our objets and when we use them.
-
-### Private by default
+```
+begin
+  require_relative "heroku_buildpack_wrapped_rakefile.rb"
+rescue => e
+  STDERR.puts <<~EOM
+    Error caught and ignored while loading Rakefile
 
-Methods and accessors should be private when possible. This allows for a more explicit API. An explicit API provides more room for future refacoring and changes while minimizing changes needed to other classes. You can make attr_ methods private like this:
+    To fail the build instead, remove the buildpack TODO
 
-```ruby
-class Whatever
-  private; attr_reader :foo, :bar; public
+    Message:
+    #{e.message}
+  EOM
 end
 ```
 
-### attr_reader for values that may be falsey
+### Known differences against `heroku/heroku-buildpack-ruby`
 
-When working with a variable that may be falsey, prefer to use an `attr_reader` over accessing the instance variable directly. This will guard against spelling mistakes. For example:
+This buildpack does not port all behaviors of the original buildpack for Ruby (https://github.com/heroku/heroku-buildpack-ruby). This buildpack is also known as `v2` as it implements version 2 of the heroku buildpack interface (instead of the Cloud Native Buildpack interface).
 
-```ruby
-class Foo
-  private; attr_reader :value; public
+- Rails 5+ support only. The v2 buildpack supports Rails 2+. There are significant maintenace gains for buildpack authors [starting in Rails 5](https://blog.heroku.com/container_ready_rails_5) which was released in 2016. In an effort to reduce overall internal complexity this buildpack does not explicitly support Rails before version 5.
+- Rails support is now based on the above application contract. Previously there was no official policy for dropping support for older Rails versions. Support was based on whether or not an older version could run on a currently supported Ruby version. With [Rails LTS](https://railslts.com/) this can mean a very old version of Rails. Now we will actively support Rails versions currently [under the Rails core maintenance policy](https://guides.rubyonrails.org/maintenance_policy.html) provided those versions can run on a supported Ruby version. As Rails versions go out of official maintenance compatability features may be removed with no supported replacement.
```

Comment:
```suggestion
- Rails support is now based on the above application contract. Previously there was no official policy for dropping support for older Rails versions. Support was based on whether or not an older version could run on a currently supported Ruby version. With [Rails LTS](https://railslts.com/) this can mean a very old version of Rails. Now we will actively support Rails versions currently [under the Rails core maintenance policy](https://guides.rubyonrails.org/maintenance_policy.html) provided those versions can run on a supported Ruby version. As Rails versions go out of official maintenance compatibility features may be removed with no supported replacement.
```